moped-database/migrations/1651692507000_alter_project_list_view_type_bug/up.sql

```diff
@@ -0,0 +1,111 @@
+DROP VIEW project_list_view;
+
+CREATE VIEW public.project_list_view AS
+with project_person_list_lookup as (
+  SELECT
+    mpp.project_id,
+    string_agg(DISTINCT concat(mu.first_name, ' ', mu.last_name, ':', mpr.project_role_name), ',') AS project_team_members
+   FROM 
+    public.moped_proj_personnel mpp
+        JOIN public.moped_users mu ON mpp.user_id = mu.user_id
+        JOIN public.moped_project_roles mpr ON mpp.role_id = mpr.project_role_id
+      WHERE (mpp.status_id = 1)
+      GROUP BY mpp.project_id
+),
+ funding_sources_lookup as (
+  SELECT
+    mpf.project_id,
+    string_agg(mfs.funding_source_name, ', ') as funding_source_name
+  FROM
+    moped_proj_funding mpf
+    LEFT JOIN moped_fund_sources mfs ON mpf.funding_source_id = mfs.funding_source_id
+    GROUP BY mpf.project_id
+),
+project_type_lookup as (
+  SELECT
+    mpt.project_id,
+    string_agg(mt.type_name, ', ') as type_name
+  FROM
+     public.moped_project_types mpt
+     LEFT JOIN public.moped_types mt ON mpt.project_type_id = mt.type_id AND mpt.status_id = 1
+     GROUP BY mpt.project_id
+)
```

Comment:
Here is the fix for the bug. Its a one to many relationship, I was previously treating it as a one to one. Also if someone removes a project type, its a soft delete, so we need to only get the ones that are active. 

I don't think we need to maintain it as a soft delete, I'm open to discussing if it should stay as a soft delete or if we should change it.

Reply:
(lines 24-32)

Reply:
By my read, you're building up a CTE with two columns (project ID and list of types) and using that as a lookup table to get that list associated to the projects being returned as rows in the main query. 👍 I think using a CTE here is both clear in terms of readability and fast as it avoids any unnecessary looping in the query plan. 

Testing this as-written, without moving the sub-select up into the select clause as suggested in the review to #642, still yields a query that executes in ~30 ms, which is great.

Reply:
I don't really have an opinion on if this specific case should be a soft delete. It seems like we adopted that `status_id=1` pattern all over the place. And we all agree now that a better approach would be a boolean value column specifically for soft deletes. But the more we add in conditions like this, buried in a migration file, the harder it may be for us to refactor away from this pattern in the future. No solutions, just thoughts.